pandas/core/indexes/period.py

```diff
@@ -350,17 +530,28 @@ def asof_locs(self, where, mask):
 
         return result
 
+    def _box_values_as_index(self):
```

Comment:
this looks like a new name, do we really need this?

Reply:
See the todo, it can be removed once the split is complete (also the other Datetime-like index/array are splitted). Or at least that is how I understand the comment.

Reply:
Well, apparently I changed the behavior. On master we return an object-dtype `Index[Period]`, but on this PR it returns a `PeriodIndex`. I'll try removing.

Reply:
Seems ok to remove.